notes/serializers/serializers.py

```diff
@@ -19,24 +37,100 @@ class Meta:
         model = Note
         fields = [
             "id",
-            "title",
-            "text",
             "user",
             "date_created",
             "date_updated",
+            "title",
+            "text",
             "shared_with",
-            "archive",
+            "comments",
         ]
         indexes = [
             models.Index(fields=["date_updated", "-date_created"]),
         ]
+        extra_kwargs = {"title": {"required": True}, "text": {"required": True}}
+        ordering = ["-id"]
+
+    def update(self, instance, validated_data):
+        NoteVersion.objects.create(
+            note_id=instance,
+            title=instance.title,
+            text=instance.text,
+            edited_by=self.context["request"].user,
+            date_created=instance.date_created,
+            date_updated=instance.date_updated,
+        )
+
+        return super().update(instance, validated_data)
+
+    def to_representation(self, instance):
+        """
+        Overriding this function to just show the latest comment if we list
+        down all the notes of a user
+        """
+
+        representation = super().to_representation(instance)
+        if representation["comments"]:
```

Comment:
Please improve the logic for getting the last comment. You can improve it by just fetching the last comment in the queryset.